### PR TITLE
move utils from pipcook-core to single package

### PR DESCRIPTION
**Pull request by utkobe**
This is to decouple pipcook-core which contains types definition of pipcook with any utils functions that dependent on nodejs module. So that pipcook-core can also be used in browser (pipboard).

### CODE REVIEW

File: packages/cli/package.json [hunk 1]
```diff
@@ -37,7 +37,8 @@
     "request": "^2.88.2",
     "semver": "^6.3.0",
     "ts-node": "^8.9.1",
-    "yaml": "^1.7.2"
+    "yaml": "^1.7.2",
+    "@pipcook/pipcook-utils": "^0.5.0"
```
Comment (yorkie) [inline packages/cli/package.json:41]: Sort the list in alphabeta :)

File: packages/cli/src/commands/serve.ts [hunk 2]
```diff
@@ -2,13 +2,12 @@ import Fastify from 'fastify';
 import path from 'path';
 import ora from 'ora';
 import childProcess from 'child_process';
-import { constants } from '@pipcook/pipcook-core';
+import { PIPCOOK_LOGS } from '@pipcook/pipcook-utils';
```
Comment (yorkie) [inline packages/cli/src/commands/serve.ts:5]: Constants should be in core?

File: packages/cli/src/config.ts [hunk 3]
```diff
@@ -1,3 +1,6 @@
+import * as path from "path";
```
Comment (yorkie) [inline packages/cli/src/config.ts:1]: Never used.

File: packages/cli/src/config.ts [hunk 4]
```diff
@@ -1,3 +1,6 @@
...
+import * as os from 'os';
```
Comment (yorkie) [inline packages/cli/src/config.ts:2]: Ditto.

File: packages/daemon/package.json [hunk 5]
```diff
@@ -12,6 +12,7 @@
     "uuid": "^8.0.0",
     "uuid-validate": "^0.0.3",
     "@pipcook/pipcook-core": "^0.5.16",
+    "@pipcook/pipcook-utils": "^0.5.0",
```
Comment (yorkie) [inline packages/daemon/package.json:15]:
```suggestion
    "@pipcook/utils": "^0.5.0",
```

File: packages/plugins/data-collect/image-classification-data-collect/package.json [hunk 6]
```diff
@@ -21,7 +21,8 @@
     "fs-extra": "^8.1.0",
     "glob": "^7.1.6",
     "glob-promise": "^3.4.0",
-    "uuid": "^7.0.2"
+    "uuid": "^7.0.2",
+    "@pipcook/pipcook-utils": "^0.5.0"
```
Comment (yorkie) [inline packages/plugins/data-collect/image-classification-data-collect/package.json:25]: IMO the plugin won't depend on core.

File: packages/utils/package.json [hunk 7]
```diff
@@ -0,0 +1,44 @@
+{
+  "name": "@pipcook/pipcook-utils",
```
Comment (yorkie) [inline packages/utils/package.json:2]:
```suggestion
  "name": "@pipcook/utils",
```

File: packages/utils/src/constants.ts [hunk 8]
```diff
@@ -0,0 +1,7 @@
+import * as os from 'os';
+import * as path from 'path';
+
+export const PIPCOOK_HOME_PATH = path.join(os.homedir(), '.pipcook');
+export const PIPCOOK_PLUGINS = path.join(PIPCOOK_HOME_PATH, 'plugins');
+export const PIPCOOK_LOGS = path.join(PIPCOOK_HOME_PATH, 'logs');
+export const PIPCOOK_DEPENDENCIES = path.join(PIPCOOK_HOME_PATH, 'dependencies');
```
Comment (Txiaozhe) [inline packages/utils/src/constants.ts:7]: How about aggregate these paths to a named constant such as 'PIPCOOK_PATH'?

File: packages/cli/src/commands/serve.ts (hunk 2 above)
Comment (utkobe) [reply packages/cli/src/commands/serve.ts:5]: These constants are path related which requires path module. So I move them into utils

File: packages/plugins/data-collect/image-classification-data-collect/package.json (hunk 6 above)
Comment (utkobe) [reply packages/plugins/data-collect/image-classification-data-collect/package.json:25]: What's the reason. Core provides some definitions of types of plugins

File: packages/utils/src/constants.ts (hunk 8 above)
Comment (utkobe) [reply packages/utils/src/constants.ts]: Good idea. Fixed

File: packages/utils/package.json (hunk 7 above)
Comment (utkobe) [reply packages/utils/package.json]: fixed

File: packages/daemon/package.json (hunk 5 above)
Comment (utkobe) [reply packages/daemon/package.json]: fixed

File: packages/cli/src/config.ts (hunk 4 above)
Comment (utkobe) [reply packages/cli/src/config.ts]: fixed

File: packages/cli/src/config.ts (hunk 3 above)
Comment (utkobe) [reply packages/cli/src/config.ts]: fixed

File: packages/cli/package.json (hunk 1 above)
Comment (utkobe) [reply packages/cli/package.json]: fixed

File: packages/plugins/data-collect/image-classification-data-collect/package.json (hunk 6 above)
Comment (yorkie) [reply packages/plugins/data-collect/image-classification-data-collect/package.json]: IMO, the core will be our internal lib, and we need to keep fucking simple for plugin developers.

File: packages/cli/src/commands/serve.ts (hunk 2 above)
Comment (yorkie) [reply packages/cli/src/commands/serve.ts]: As I have thought at https://github.com/alibaba/pipcook/pull/195#discussion_r429023922, if the core is used internally, it depends on if we need to provide these paths to plugin developers.

**Pull request closed without merging**